CONTRIBUTING.md

```diff
@@ -1,47 +1,297 @@
 
-## Getting started
+## Pre submit flight checks
 
-- Fork the repository on GitHub.
-- Read the README.md for build instructions.
-- Play with the project, submit bugs, submit patches!
+Make sure you decide whether your issue and/or pull request is improving TiDB architecture or whether its simply fixing a bug.
 
-## Contribution flow
+Make sure there are no typos, if you need a diagram, add it.  Make sure you SEPARATE the description of the problem
+(i.e. Y is a critical component that is too slow for an SLA that we care about) from the solution (i.e. make X faster).
 
-This is a rough outline of what a contributor's workflow looks like:
+In order to save reviewers time, each issue should be file with card and should be sanity-checkable in under 5 minutes(even 
+the busiest of reviewers can spare up to 5 mintutes to review a patch that is thoughtfully justified)
 
-- Create a topic branch from where you want to base your work. This is usually master.
-- Make commits of logical units and add test case if the change fixes a bug or adds new functionality.
-- Run tests and make sure all the tests are passed.
-- Make sure your commit messages are in the proper format (see below).
-- Push your changes to a topic branch in your fork of the repository.
-- Submit a pull request to pingcap/tidb.
-- Your PR must receive LGTMs from two maintainers found in the [MAINTAINERS](./docs/MAINTAINERS.md) file.
+### Is this just a simple bug fix?
 
-Thanks for your contributions!
+These patches can be easy to review since test coverage is submitted with the patch.  Bug fixes don't usually require a lot
+of extra testing: But please update the unit tests so that they catch the bug !
 
-### Code style
+### Is this an architecture improvement?
 
-The coding style suggested by the Golang community is used in TiDB. See the [style doc](https://github.com/golang/go/wiki/CodeReviewComments) for details.
+Some examples of "Architecture" improvements:
 
-Please follow this style to make TiDB easy to review, maintain and develop.
+- Adding a new feature or making a feature more configurable/modular.
+- Converting structs to interfaces.
+- Improving test coverage.
+- Decoupling logic or creation of new utilities.
+- Making code more resilient (sleeps, backoffs, reducing flakiness, etc).
+
+These sorts of improvements are easily evaluated if they decrease lines of code without breaking functionality.
+
+If you are improving the quality of code, then justify/state exactly what you 'cleaning up' in your Pull Request so as
+not to waste reviewer time.
+
+If you're making code more resilient, test it locally to demonstrate how exactly your patch changes
+things.
+
+### Is this a performance improvement ?
+
+If you are submitting a performance bug, you MUST ALSO submit data that demonstrates your problem if you want the issue to
+remain open.  This can be done locally using go benchmark tests. Such improvements must come with benchmark test which convince reviewers it is 
```

Comment:
because i think that you want to emphasize the result ("convince reviewers...") and which-clause highlights something supplementary, which in this case makes readers think that the convince thing refers to the improvements themselves.